conftest.py

```diff
@@ -1,8 +0,0 @@
-import pkg_resources
-
-entry_points = []
-for entry_point in pkg_resources.iter_entry_points('pytest11'):
-    entry_points.append(entry_point.name)
-
-if "asdf_schema_tester" not in entry_points:
-    pytest_plugins = ['asdf.tests.schema_tester']
```

Comment:
This appears to be attempting to register the asdf schema testing part of the asdf pytest plugin. As gwcs no longer contains schema files this is no longer needed.

Reply:
Removing this file entirely breaks `test-pyargs` for reasons unknown to me.

With the empty `conftest.py` pytest finds tests at paths like the following:
```
../gwcs/tests/test_wcs.py ...........................................s.. [ 97%]
.s...s....s..ss........                                                  [100%]
```
Without `conftest.py` pytest will collect tests with slightly different paths:
```
../.tox/test-pyargs/lib/python3.11/site-packages/gwcs/tests/test_wcs.py . [ 93%]
.................F......EEEEEEEEEEF.......s...s...s....s..ssFF.F..FF     [100%]
```
and some tests (those that access http://www.astropy.org/astropy-data/data/) fail:
```
E               urllib.error.URLError: <urlopen error Unable to open any source! Exceptions were {'http://data.astropy.org/data/acs.hdr': <HTTPError 404: 'Not Found'>, 'http://www.astropy.org/astropy-data/data/acs.hdr': <HTTPError 404: 'Not Found'>}>
```

Reply:
These failures match failures seen when running `pytest --pyargs gwcs` after installing `gwcs` from pypi.

I removed the empty `conftest.py` from this PR so the `test-pyargs` job should fail (as it appears to be intended to test if `pytest --pyargs gwcs` would fail).

Reply:
The failure appears to be caused by `data` not be included in the released package.